[FEATURE] Add information on how to use the submodule on your extension

# Feature Request:

Could you add detailed information on how to work on the extension? 

My first hurdle was to figure out, that `git clone` wouldn't clone the `whats-new` section. And even after that I couldn't get it to run


# Background / Details:

In regard of [this](https://github.com/alefragnani/vscode-separators/issues/21) issue, I thought, I'd look into your code base. But I have troubles debugging your extension. Apparently it has troubles finding the what's new section.

```
Activating extension 'alefragnani.separators' failed: Cannot find module '../../vscode-whats-new/src/Manager'.
```

--- a/README.md
+++ b/README.md
@@ -17,6 +17,13 @@ You just need to add a `submodule` reference to your Git repo
 
 ### Coding
 
+This submodule is based on a `Provider` pattern (similar to some pieces of VS Code), and you need to implement at least the `ContentProvider` in order to display the main content of the page, like the Header (logo and message) and Changelog.
+
+There are other two providers available, for specific contents:
+
+* `SocialMediaProvider`": Add links to your social medias
+* `SponsorProvider`: Add references (logos and messages) to your sponsors.
+
 ```ts
 
 // provide the data

--- a/README.md
+++ b/README.md
@@ -1,10 +1,21 @@
 # What's New submodule
 
-This submodule is used to display a **What's New** page on my VS Code extensions. It has a simple (but yet effective) UI, optimized to display all the information in a single page.
+This submodule is used to display a **What's New** page on VS Code extensions. It has a simple (but yet effective) UI, optimized to display all the information in a single page.
 
 ![whats-new-bookmarks](images/whats-new-bookmarks.png)
 
 ## Usage
+
+### Adding
+
+You just need to add a `submodule` reference to your Git repo
+
+```
+    git submodule add https://github.com/alefragnani/vscode-whats-new.git
+
+```
+
+### Coding
 
 ```ts
 

